pkg/extensions/sync/remote.go

```diff
@@ -50,13 +51,33 @@ func (registry *RemoteRegistry) GetContext() *types.SystemContext {
 func (registry *RemoteRegistry) GetRepositories(ctx context.Context) ([]string, error) {
 	var catalog catalog
 
-	_, _, _, err := registry.client.MakeGetRequest(ctx, &catalog, "application/json", //nolint: dogsled
+	_, header, _, err := registry.client.MakeGetRequest(ctx, &catalog, "application/json", "", //nolint: dogsled
 		constants.RoutePrefix, constants.ExtCatalogPrefix)
 	if err != nil {
 		return []string{}, err
 	}
 
-	return catalog.Repositories, nil
+	repos := catalog.Repositories
```

Comment:
I think simply constructing an empty array and keep appending to that should do the trick:

```suggestion
	var repos []string

	repos = append(repos, catalog.Repositories...)
```